pex/cli/commands/cache/command.py

```diff
@@ -0,0 +1,391 @@
+                argument=self,
+                message=reportable_unexpected_error_msg(
+                    "The HandleAmountAction was used in an unexpected place."
+                ),
+            )
+        setattr(namespace, self.dest, amount_func)
+
+
+class Cache(OutputMixin, BuildTimeCommand):
+    """Interact with the Pex cache."""
+
+    @staticmethod
+    def _add_amount_argument(parser):
+        # type: (_ActionsContainer) -> None
+
+        parser.add_argument(
+            "-H",
+            "--human-readable",
+            *["--{unit}".format(unit=unit.value) for unit in ByteUnits.values()],
+            dest="amount_func",
+            action=HandleAmountAction,
+            default=ByteAmount.bytes,
+            help=(
+                "How to display disk usage amounts; defaults to bytes. The -H / --human-readable "
+                "options display amounts in a human readable way and each of the other unit "
+                "options displays amounts with that specified unit."
+            )
+        )
+
+    @classmethod
+    def _add_info_arguments(cls, parser):
+        # type: (_ActionsContainer) -> None
+
+        cls._add_amount_argument(parser)
+        parser.add_argument(
+            "-S",
+            "--sort-size",
+            dest="sort_by_size",
+            action="store_true",
+            help=(
+                "Don't actually purge cache entries; instead, perform a dry run that just prints "
+                "out what actions would be taken"
+            ),
+        )
+        parser.add_argument(
+            "-r",
+            "--reverse",
+            dest="reverse",
+            action="store_true",
+            help=(
+                "Don't actually purge cache entries; instead, perform a dry run that just prints "
+                "out what actions would be taken"
+            ),
+        )
+        cls.add_output_option(parser, entity="Pex cache information")
+
+    @classmethod
+    def _add_purge_arguments(cls, parser):
+        # type: (_ActionsContainer) -> None
+
+        cls._add_amount_argument(parser)
+        parser.add_argument(
+            "--entries",
+            action="append",
+            type=CacheDir.for_value,
+            choices=CacheDir.values(),
+            default=[],
+            help=(
+                "Specific cache entries to purge. By default, all entries are purged, but by "
+                "specifying one or more particular --entries to purge, only those entries (and any "
+                "other cache entries dependent on those) will be purged."
+            ),
+        )
+        parser.add_argument(
+            "-n",
+            "--dry-run",
+            dest="dry_run",
+            action="store_true",
+            help=(
+                "Don't actually purge cache entries; instead, perform a dry run that just prints "
+                "out what actions would be taken"
+            ),
+        )
+        parser.add_argument(
```

Comment:
Why wouldn't we always emit this?

Reply:
Collecting du with a cold page cache is slow.

That said, it should be roughly as slow as the rm. Since both do the same fs walk in close succession, it seems the second, rm walk only adds ~20% instead of doubling the time. Here I test with a ~1GB Pex cache:
```console
:; hyperfine -p 'rm -rf ~/.cache/pex && cp -r ~/.cache/pex-bak ~/.cache/pex && free && sync && sudo sh -c "echo 3 > /proc/sys/vm/drop_caches" && free && sleep 2' 'python -mpex.cli cache purge' 'pytho
n -mpex.cli cache purge -RH'
Benchmark 1: python -mpex.cli cache purge
  Time (mean ± σ):     851.7 ms ±  14.0 ms    [User: 327.0 ms, System: 711.2 ms]
  Range (min … max):   826.7 ms … 873.2 ms    10 runs

Benchmark 2: python -mpex.cli cache purge -RH
  Time (mean ± σ):     998.0 ms ±   8.4 ms    [User: 460.2 ms, System: 811.0 ms]
  Range (min … max):   986.5 ms … 1017.9 ms    10 runs

Summary
  python -mpex.cli cache purge ran
    1.17 ± 0.02 times faster than python -mpex.cli cache purge -RH
```

I guess having to wait 6 minutes instead of 5 for a ~300GB Pex cache is good enough for a 1st release. I can add the inverse `--no-report` later if needed.